intrastat_base/intrastat_common.py

```diff
@@ -0,0 +1,167 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Report intrastat base module for Odoo
+#    Copyright (C) 2010-2014 Akretion (http://www.akretion.com/).
+#    @author Alexis de Lattre <alexis.delattre@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, tools, _
+from openerp.exceptions import Warning, ValidationError
+from dateutil.relativedelta import relativedelta
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ReportIntrastatCommon(models.AbstractModel):
+    _name = "report.intrastat.common"
+    _description = "Common functions for intrastat reports for products "
+    "and services"
+
+    @api.one
+    @api.depends(
+        'intrastat_line_ids', 'intrastat_line_ids.amount_company_currency')
```

Comment:
You can depends only on `intrastat_line_ids.amount_company_currency` as it implies `intrastat_line_ids`